renderer.cpp

```diff
@@ -1,19 +1,43 @@
 #include "renderer.h"
 
+#include "indexbuffer.h"
 #include "renderutils.h"
+#include "texture.h"
+#include "vertexarray.h"
+
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
 
 #include <iostream>
 
-void Renderer::Draw(const VertexArray& va, const IndexBuffer& ib, const Shader& shader) const
+std::unique_ptr<Renderer> Renderer::sInstance = std::make_unique<Renderer>();
```

Comment:
You could consider making this a meyer's singleton. The pattern looks like this:

```
class Singleton
{
  
public:
  static Singleton& Instance()
  {
    static Singleton s; 
    return s;
  }
  
private:
  Singleton() {}
  Singleton(const Singleton &) = delete;
  Singleton& operator= (const Singleton &) = delete;
  ~Singleton() {}

};
```

Then all access, like on line 17 looks like this:

```
instance.mShader = make_unique<Shader>...
```


Note a few benefits of this: The instance is instantiated the first time it is used, not somewhere in the insanity of static initialization. Next, you aren't dealing with pointers.

Reply:
Yeah that is cooler. Although I couldn't access the instance just through name like you showed, I had to use `GetInstance()`. I figured that is what you meant, or that I set it up wrong.

Reply:
Yep, my typo. You did it as I intended to write it 👍